nixos/modules/services/databases/postgresql.nix

```diff
@@ -428,7 +454,43 @@ in
           PostgreSQL superuser account to use for various operations. Internal since changing
           this value would lead to breakage while setting up databases.
         '';
-        };
+      };
+
+      upgradeFrom = mkOption {
+        type = types.nullOr (types.submodule ({ config, ... }: {
+          options = {
+            package = mkOption {
+              type = types.package;
+              description = lib.mdDoc "Package to upgrade from";
+            };
+            dataDir = mkOption {
+              type = types.path;
+              default = "/var/lib/postgresql/${config.package.psqlSchema}";
+              defaultText = literalExpression
+                ''"/var/lib/postgresql/''${config.package.psqlSchema}"'';
```

Comment:
```suggestion
              default = config.services.postgresql.dataDir;
              defaultText = literalExpression "config.services.postgresql.dataDir";
```

Reply:
Wait, this is where the prior postgres data dir lives, so it should be `/var/lib/postgresql/<prior version>`, right?

Reply:
(this config refers to the submodule config)

Reply:
Perhaps no default would be better...

Reply:
Defaulted using `stateVersion`.